In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt 
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

In [2]:
df = pd.read_csv("C:/Users/loye/Desktop/FoodBalanceSheets_E_Africa_NOFLAG.csv", encoding = "latin1")

In [3]:
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [4]:
df.shape

(60943, 12)

In [5]:
df.describe(include = 'all')

Area Code   Area     Item Code                     Item  \
count   60943.000000  60943  60943.000000                    60943   
unique           NaN     49           NaN                      119   
top              NaN  Kenya           NaN  Milk - Excluding Butter   
freq             NaN   1560           NaN                     1262   
mean      134.265576    NaN   2687.176706                      NaN   
std        72.605709    NaN    146.055739                      NaN   
min         4.000000    NaN   2501.000000                      NaN   
25%        74.000000    NaN   2562.000000                      NaN   
50%       136.000000    NaN   2630.000000                      NaN   
75%       195.000000    NaN   2775.000000                      NaN   
max       276.000000    NaN   2961.000000                      NaN   

        Element Code                   Element         Unit          Y2014  \
count   60943.000000                     60943        60943   59354.000000   
unique           NaN                        18            5            NaN   
top              NaN  Domestic supply quantity  1000 tonnes            NaN   
freq             NaN                      5295        40933            NaN   
mean     3814.856456                       NaN          NaN     134.196282   
std      2212.007033                       NaN          NaN    1567.663696   
min       511.000000                       NaN          NaN   -1796.000000   
25%       684.000000                       NaN          NaN       0.000000   
50%      5142.000000                       NaN          NaN       0.090000   
75%      5511.000000                       NaN          NaN       8.340000   
max      5911.000000                       NaN          NaN  176405.000000   

                Y2015          Y2016          Y2017          Y2018  
count    59395.000000   59408.000000   59437.000000   59507.000000  
unique            NaN            NaN            NaN            NaN  
top               NaN            NaN            NaN            NaN  
freq              NaN            NaN            NaN            NaN  
mean       135.235966     136.555222     140.917765     143.758381  
std       1603.403984    1640.007194    1671.862359    1710.782658  
min      -3161.000000   -3225.000000   -1582.000000   -3396.000000  
25%          0.000000       0.000000       0.000000       0.000000  
50%          0.080000       0.080000       0.100000       0.070000  
75%          8.460000       8.430000       9.000000       9.000000  
max     181137.000000  185960.000000  190873.000000  195875.000000

In [6]:
df.dtypes

Area Code         int64
Area             object
Item Code         int64
Item             object
Element Code      int64
Element          object
Unit             object
Y2014           float64
Y2015           float64
Y2016           float64
Y2017           float64
Y2018           float64
dtype: object

#### Checking for missing columns

In [7]:
for column in df.isnull().columns.values.tolist():
    print(column)
    print (df.isnull()[column].value_counts())
    print("")   

Area Code
False    60943
Name: Area Code, dtype: int64

Area
False    60943
Name: Area, dtype: int64

Item Code
False    60943
Name: Item Code, dtype: int64

Item
False    60943
Name: Item, dtype: int64

Element Code
False    60943
Name: Element Code, dtype: int64

Element
False    60943
Name: Element, dtype: int64

Unit
False    60943
Name: Unit, dtype: int64

Y2014
False    59354
True      1589
Name: Y2014, dtype: int64

Y2015
False    59395
True      1548
Name: Y2015, dtype: int64

Y2016
False    59408
True      1535
Name: Y2016, dtype: int64

Y2017
False    59437
True      1506
Name: Y2017, dtype: int64

Y2018
False    59507
True      1436
Name: Y2018, dtype: int64



## Question 11

In [8]:
df_item = df.groupby(['Item']).sum()
df_item.loc['Animal fats']

Area Code         85279.00
Item Code       1882494.00
Element Code    2567303.00
Y2014            209460.54
Y2015            200675.72
Y2016            183314.22
Y2017            269617.53
Y2018            269648.27
Name: Animal fats, dtype: float64

## Question 12

In [9]:
df_mean = df[['Y2015']].mean().round(3)
print('The mean value for',df_mean)

The mean value for Y2015    135.236
dtype: float64


In [10]:
df_std = df[['Y2015']].std().round(3)
print('The standard deviation value for',df_std)

The standard deviation value for Y2015    1603.404
dtype: float64


## Question 13

In [11]:
print('The number of missing values for',df[['Y2016']].isnull().sum())

The number of missing values for Y2016    1535
dtype: int64


In [12]:
percentage_missing = df[['Y2016']].isnull().sum()*100/len(df[['Y2016']])

In [13]:
# rounding percentage of missing values to 2 decimal places
percentage_round = round(percentage_missing, 2)
percentage_round

Y2016    2.52
dtype: float64

## Question 14

In [14]:
df.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

#### The year 2014 had the highest correlation value with Element Code with a value of 0.024457

## Question 15

In [15]:
df_element = df.groupby(['Element']).sum()
df_element.loc['Import Quantity']

Area Code         688174.00
Item Code       13795966.00
Element Code    28834929.00
Y2014             274144.48
Y2015             267018.46
Y2016             286582.78
Y2017             294559.09
Y2018             287997.09
Name: Import Quantity, dtype: float64

#### The year 2017 had the highest sum for Import Quantity

## Question 16

In [16]:
df_element.loc['Production']

Area Code         526751.00
Item Code       10450053.00
Element Code    21388191.00
Y2014            1931287.75
Y2015            1947019.39
Y2016            1943537.15
Y2017            2030056.89
Y2018            2075072.89
Name: Production, dtype: float64

#### The year 2014 had a total sum of 1931287.75

## Question 17

In [17]:
df_element['Y2018']

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

#### Domestic supply quantity had the highest sum in 2018

## Question 18

#### Protein supply quantity (g/capita/day) had the third lowest figure

## Question 19

In [18]:
df_area = df.groupby(['Area', 'Element']).sum()
df_area

Area Code  Item Code  \
Area     Element                                                        
Algeria  Domestic supply quantity                      464     311204   
         Export Quantity                               392     264011   
         Fat supply quantity (g/capita/day)            432     291489   
         Feed                                           88      58967   
         Food                                          420     282744   
...                                                    ...        ...   
Zimbabwe Protein supply quantity (g/capita/day)      20453     304589   
         Residuals                                   18824     277476   
         Seed                                         3439      50560   
         Stock Variation                             17195     254157   
         Total Population - Both sexes                 181       2501   

                                                 Element Code     Y2014  \
Area     Element                                                          
Algeria  Domestic supply quantity                      614916  83755.94   
         Export Quantity                               579278   1328.64   
         Fat supply quantity (g/capita/day)             73872    320.19   
         Feed                                          121462   9656.04   
         Food                                          539910  60582.69   
...                                                       ...       ...   
Zimbabwe Protein supply quantity (g/capita/day)         76162    184.23   
         Residuals                                     537680   -252.00   
         Seed                                          105013    144.00   
         Stock Variation                               481840   -644.00   
         Total Population - Both sexes                    511  13587.00   

                                                    Y2015     Y2016     Y2017  \
Area     Element                                                                
Algeria  Domestic supply quantity                85972.86  87157.61  86810.42   
         Export Quantity                          1104.62   1340.22   1494.54   
         Fat supply quantity (g/capita/day)        317.40    309.54    320.74   
         Feed                                    10308.00   9903.00   9571.00   
         Food                                    61693.49  63081.43  62785.33   
...                                                   ...       ...       ...   
Zimbabwe Protein supply quantity (g/capita/day)    175.31    172.72    175.31   
         Residuals                                -780.00   -882.00  -1083.00   
         Seed                                      128.00    127.00    162.00   
         Stock Variation                          -343.00    221.00    458.00   
         Total Population - Both sexes           13815.00  14030.00  14237.00   

                                                    Y2018  
Area     Element                                           
Algeria  Domestic supply quantity                90023.42  
         Export Quantity                           811.54  
         Fat supply quantity (g/capita/day)        309.09  
         Feed                                    10668.00  
         Food                                    63736.33  
...                                                   ...  
Zimbabwe Protein supply quantity (g/capita/day)    176.57  
         Residuals                                -546.00  
         Seed                                      140.00  
         Stock Variation                          -447.00  
         Total Population - Both sexes           14439.00  

[803 rows x 8 columns]

In [19]:
df_area.loc['Algeria']

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                      464     311204        614916   
Export Quantity                               392     264011        579278   
Fat supply quantity (g/capita/day)            432     291489         73872   
Feed                                           88      58967        121462   
Food                                          420     282744        539910   
Food supply (kcal/capita/day)                 432     291489         71712   
Food supply quantity (kg/capita/yr)           420     282744         67725   
Import Quantity                               448     301030        628432   
Losses                                        208     139857        266396   
Other uses (non-food)                         148     101269        190698   
Processing                                    140      93537        179585   
Production                                    360     242590        495990   
Protein supply quantity (g/capita/day)        432     291489         72792   
Residuals                                     408     272327        527340   
Seed                                           80      53109        110540   
Stock Variation                               376     251696        476768   
Total Population - Both sexes                   4       2501           511   

                                           Y2014     Y2015     Y2016  \
Element                                                                
Domestic supply quantity                83755.94  85972.86  87157.61   
Export Quantity                          1328.64   1104.62   1340.22   
Fat supply quantity (g/capita/day)        320.19    317.40    309.54   
Feed                                     9656.04  10308.00   9903.00   
Food                                    60582.69  61693.49  63081.43   
Food supply (kcal/capita/day)           13500.00  13512.00  13483.00   
Food supply quantity (kg/capita/yr)      1556.45   1552.81   1555.77   
Import Quantity                         34940.12  37841.91  36736.08   
Losses                                   7231.00   7471.00   7405.00   
Other uses (non-food)                    3315.22   3483.37   3400.17   
Processing                               1776.00   1865.00   1920.00   
Production                              52534.48  53451.57  54554.75   
Protein supply quantity (g/capita/day)    379.56    377.37    378.84   
Residuals                                 552.00    489.00    770.00   
Seed                                      631.00    657.00    672.00   
Stock Variation                          2392.00   4233.00   2789.00   
Total Population - Both sexes           38924.00  39728.00  40551.00   

                                           Y2017     Y2018  
Element                                                     
Domestic supply quantity                86810.42  90023.42  
Export Quantity                          1494.54    811.54  
Fat supply quantity (g/capita/day)        320.74    309.09  
Feed                                     9571.00  10668.00  
Food                                    62785.33  63736.33  
Food supply (kcal/capita/day)           13363.00  13288.00  
Food supply quantity (kg/capita/yr)      1516.94   1509.28  
Import Quantity                         36059.29  36238.29  
Losses                                   7428.00   7708.00  
Other uses (non-food)                    3534.08   3970.08  
Processing                               2045.00   2479.00  
Production                              54869.67  60766.67  
Protein supply quantity (g/capita/day)    371.26    367.28  
Residuals                                 766.00    630.00  
Seed                                      684.00    825.00  
Stock Variation                          2636.00   6172.00  
Total Population - Both sexes           41389.00  42228.00

#### The total for import quantity for Algeria in 2018 is 36238.29

## Question 20

In [20]:
df[['Area']].nunique()

Area    49
dtype: int64

#### The total number of unique countries is 49